lib/validators/specValidator.ts

```diff
@@ -159,7 +162,7 @@ export class SpecValidator<T extends CommonValidationResult> {
    * Initializes the spec validator. Resolves the spec on different counts using the SpecResolver
    * and initializes the internal api validator.
    */
-  public async initialize(): Promise<Sway.SwaggerApi> {
+  public readonly initialize = async (): Promise<Sway.SwaggerApi> => {
```

Comment:
any advantage of this ? although of fan of arrow functions, i tend to avoid them for class instance methods for a couple of reasons:
   - harder to mock
   - breaks inheritance
  - performance goes down dramatically because the functions are not in the prototype

Reply:
That's the only way (I've found) how to make the function immutable. https://github.com/ts-common/ts-common.github.io/blob/master/examples/functions.ts

Reply:
Please, no inheritance :-)

Reply:
given the downsides, any specific reason we want to make the instance functions immutable ? What's the problem we have with that ?